source/Octopus.Server.Client/Repositories/Async/ArchivedEventFileRepository.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Octopus.Client.Model;
+using Octopus.Client.Model.Endpoints;
+using Octopus.Client.Model.EventRetention;
+
+namespace Octopus.Client.Repositories.Async
+{
+    public interface IArchivedEventFileRepository : 
+        IDelete<ArchivedEventFileResource>
+    {
+        Task<Stream> GetContent(ArchivedEventFileResource archiveEventFile);
+        Task<ResourceCollection<ArchivedEventFileResource>> List(int skip = 0, int? take = null);
+    }
+
+    class ArchivedEventFileRepository : BasicRepository<ArchivedEventFileResource>, IArchivedEventFileRepository
+    {
+        public ArchivedEventFileRepository(IOctopusAsyncRepository repository) : base(repository, "ArchivedEventFiles")
+        {
+            MinimumCompatibleVersion("2022.3.6072");
```

Comment:
Are release version are we awaiting here? GA as this is a public repo?

Reply:
This will be the version we use when rolling out this feature. 

In the plan we merge this PR after we turn this feature on for some pilot customers. So whatever version we are targeting at that point.